README.md

```diff
@@ -26,31 +26,88 @@ testProp('should detect the substring', [fc.string(), fc.string(), fc.string()],
 
 ## Advanced
 
-If you want to forward custom parameters to fast-check, `testProp` accepts an optional `fc.Parameters` ([more](https://github.com/dubzzz/fast-check/blob/master/documentation/Runners.md#runners)).
+### `fast-check` Parameters
 
-`ava-fast-check` also comes with `.only`, `.skip` and `.failing` from `ava`.
+`testProp` accepts an optional `fc.Parameters` for forwarding custom parameters to `fast-check` ([more](https://github.com/dubzzz/fast-check/blob/master/documentation/Runners.md#runners)).
 
-```javascript
+### AVA Modifiers
+
+`ava-fast-check` also comes with [`.only`], [`.serial`] [`.skip`], and [`.failing`] modifiers from AVA.
+
+```typescript
 import { testProp, fc } from 'ava-fast-check';
 
 testProp('should replay the test for the seed 4242', [fc.nat(), fc.nat()], (t, a, b) => {
   t.is(a + b, b + a);
 }, { seed: 4242 });
 
-testProp.failing('should be skipped', [fc.fullUnicodeString()], (t, text) => {
+testProp.skipped('should be skipped', [fc.fullUnicodeString()], (t, text) => {
   t.is([...text].length, text.length);
 });
 ```
 
+[`.only`]: https://github.com/avajs/ava/blob/master/docs/01-writing-tests.md#running-specific-tests
+[`.serial`]: https://github.com/avajs/ava/blob/master/docs/01-writing-tests.md#running-tests-serially
+[`.skip`]: https://github.com/avajs/ava/blob/master/docs/01-writing-tests.md#skipping-tests
+[`.failing`]: https://github.com/avajs/ava/blob/master/docs/01-writing-tests.md#failing-tests
+
+### AVA `before`/`after` Hooks
+
+`ava-fast-check` exposes AVA's `before`/`after` [hooks]:
+
+```typescript
+import { testProp, fc } from 'ava-fast-check';
+
+testProp.before(t => {
+  connectToDatabase();
+});
+
+testProp(
+  // ... omitted for brevity
+);
+
+testProp.after(t => {
+  closeDatabaseConnection();
+});
+```
+
+[hooks]: https://github.com/avajs/ava/blob/master/docs/01-writing-tests.md#before--after-hooks
+
+### AVA Execution Context
+
+`ava-fast-check` mirror's AVA's procedure for customizing the test [execution context]:
+
+```typescript
+import { fc, testProp as anyTestProp, PropertyTestInterface } from '../src/ava-fast-check';
+
+type TestContext = {
+  state: string
+};
+
+const testProp = anyTestProp as PropertyTestInterface<TestContext>;
+
+testProp(
+  'should reach terminal state',
+  [fc.string()],
+  (t, received) => {
+    // here t is typed as ExecutionContext<TestContext>
+    console.log(t.context.state); // logs 'uninitialized'
+    // ... omitted for brevity
+  }
+);
+```
+
+[execution context]: https://github.com/avajs/ava/blob/master/docs/02-execution-context.md
+
 ## Minimal requirements
 
-| ava-fast-check | ava                   | fast-check           |
+| ava-fast-check | AVA                   | fast-check           |
```

Comment:
I think the minimum-supported TypeScript version is determined by both AVA and fast-check, and not ava-fast-check, since this package is fairly simple in scope and implementation. That said, it could be useful information :+1:

Reply:
Let's go for it!

Actually some types used by ava-fast-check may restrict support of TS to only latest version. But it does not seem we rely on recent features like variadic tuples... for the moment (for instance: variadic tuples will require TS 4.x or more).